src/main/java/Duke.java

```diff
@@ -1,10 +1,165 @@
+            throw new DukeException("☹ OOPS!!! " + restOfString + " is not a valid integer for indexing the task list.");
+        } catch (IndexOutOfBoundsException err) {
+            throw new DukeException("☹ OOPS!!! There are less than " + restOfString + " tasks.");
+        }
+    }
+
+    private static void unmark(ArrayList<Task> taskList, String restOfString) throws DukeException {
+        try {
+            int i = Integer.parseInt(restOfString);
+            taskList.get(i - 1).markUnmark(false);
+        } catch (NumberFormatException err) {
+            throw new DukeException("☹ OOPS!!! " + restOfString + " is not a valid integer for indexing the task list.");
+        } catch (IndexOutOfBoundsException err) {
+            throw new DukeException("☹ OOPS!!! There are less than " + restOfString + " tasks.");
+        }
+    }
+
+    private static void addTodo(ArrayList<Task> taskList, String restOfString) throws DukeException {
+        if (restOfString.isBlank()) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        } else {
+            ToDo t = ToDo.addToDo(restOfString);
+            taskList.add(t);
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        }
+    }
+
+    private static void addDeadline(ArrayList<Task> taskList, String restOfString) throws DukeException {
+        if (restOfString.isBlank()) {
+            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+        } else {
+            Deadline t = Deadline.addDeadline(restOfString);
+            taskList.add(t);
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        }
+    }
+
+    private static void addEvent(ArrayList<Task> taskList, String restOfString) throws DukeException {
+        if (restOfString.isBlank()) {
+            throw new DukeException("☹ OOPS!!! The description of a event cannot be empty.");
+        } else {
+            Event t = Event.addEvent(restOfString);
+            taskList.add(t);
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        }
+    }
+
+    private static void delete(ArrayList<Task> taskList, String restOfString) throws DukeException {
+        try {
+            int i = Integer.parseInt(restOfString);
+            Task removedTask = taskList.remove(i-1);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(removedTask);
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        } catch (NumberFormatException err) {
+            throw new DukeException("☹ OOPS!!! " + restOfString + " is not a valid integer for indexing the task list.");
+        } catch (IndexOutOfBoundsException err) {
+            throw new DukeException("☹ OOPS!!! There are less than " + restOfString + " tasks.");
+        }
+    }
+
 }
```

Comment:
You could consider using inheritance and polymorphism for your commands, this way you wouldn't have to keep adding new cases for new commands in your `main` code. On top of this your `main` class would be a lot cleaner!